backend/src/apiserver/resource/resource_manager.go

```diff
@@ -424,6 +424,20 @@ func (r *ResourceManager) ArchiveRun(runId string) error {
 }
 
 func (r *ResourceManager) UnarchiveRun(runId string) error {
+	experimentRef, err := r.resourceReferenceStore.GetResourceReference(runId, common.Run, common.Experiment)
+	if err != nil {
+		return util.Wrap(err, "Failed to retrieve resource reference")
+	}
+
+	experiment, err := r.GetExperiment(experimentRef.ReferenceUUID)
+	if err != nil {
+		return errors.Wrap(err, "Failed to retrieve experiment")
+	}
+
+	if experiment.StorageState == api.Experiment_STORAGESTATE_ARCHIVED.String() {
+		return util.NewFailedPreconditionError(errors.New("Cannot unarchive a run that belong to archived experiment"),
+			fmt.Sprintf("Cannot unarchive run %s because it belongs to achived experiment with name `%s`.", runId, experimentRef.ReferenceName))
```

Comment:
I will be more action-driven in the error message here:

`Unarchive experiment with name %s first to allow run %s to be restored`.

Reply:
done